<?xml version="1.0" encoding="UTF-8"?>
<launch> 

	<arg name="web3_http_provider" default="http://93.187.187.109:8545" />
  	<arg name="ipfs_http_provider" default="http://127.0.0.1:5001" />

	<node pkg="dji_sdk" type="dji_sdk_node" name="dji_sdk" output="screen">
	    <!-- node parameters -->
	    <param name="serial_name" type="string" value="/dev/ttyAMA0"/>
	    <param name="baud_rate" type="int" value="230400"/>
	    <param name="app_id" type="int" value="1053855"/>
	    <param name="app_version" type="int" value="1"/>
	    <param name="align_time" type="bool" value="false"/>
	    <param name="enc_key" type="string" value="a822aa62212ee4a0f09a2e2b2538e117cf5c6710e4c3e9b675010fb596869087"/>
	    <param name="use_broadcast" type="bool" value="false"/>
    </node>

	<node name="de_airsense_mission" pkg="de_airsense" type="de_airsense_mission.py" output="screen"> </node>

	<node name="de_airsense_waspmote_ipfs" pkg="de_airsense" type="de_airsense_waspmote_ipfs.py" output="screen"> </node>

	<node name="de_airsense_agent" pkg="de_airsense" type="de_airsense_agent.py" output="screen"> 
		<param name="ipfs_http_provider" value="$(arg ipfs_http_provider)" />
	    <param name="web3_http_provider" value="$(arg web3_http_provider)" />
	    <param name="model" value="QmdFh1HPVe7H4LrDio899mxA7NindgxqiNUM9BNnBD7ryS" />
	    <param name="token" value="xrt.0.robonomics.eth" />
	    <param name="bid_lifetime" value="10" /> <!-- Ethereum blocks lifetime, ~12-15 sec for one -->
	</node>

</launch>
